### PR TITLE
[refactor] Pooling layers

### PR DESCRIPTION
**Pull request by hglee98**
## Description

This PR aims to replace pooling layers into custom `Pool` blocks.

Closes: N/A

We recommend to link at least one existing issue for PR. Before your create a PR, please check if there is an issue for this change.  

## Change(s)

- Add `POOL_REGISTRY` dict
- Add `Pool` class into `models.op.custom`
- Replace with custom `Pool` implementation

## Code Formatting

If you PR to either `master` or `dev` branch, you should follow the code linting process. Please check your code with `lint_check.sh` in `./scripts` directory.
For more information, please read the contribution guide in `CONTRIBUTING.md`. 

## Changelog

If your PR is granted to `dev` branch, codeowner will add a brief summary of the change to the **Upcoming Release** section of the [`CHANGELOG.md`](https://github.com/Nota-NetsPresso/netspresso-trainer/blob/master/CHANGELOG.md) file including a link to the PR (formatted in markdown) and a link to your github profile.

For example,

```
- Added a new feature by `@myusername` in [PR 2023](https://github.com/Nota-NetsPresso/netspresso-trainer/pull/2023)
```

Please enable **Allow edits and access to secrets by maintainers** so that our maintainers can update the `CHANGELOG.md`.

### CODE REVIEW

File: src/netspresso_trainer/models/op/custom.py [hunk 1]
```diff
@@ -172,6 +172,19 @@ def forward(self, x: Union[Tensor, Proxy]) -> Union[Tensor, Proxy]:
         return x
 
 
+class Pool(nn.Module):
```
Comment (illian01) [inline src/netspresso_trainer/models/op/custom.py:175]: I think `Pool2d` is better for clarity.

File: src/netspresso_trainer/models/op/custom.py [hunk 2]
```diff
@@ -172,6 +172,19 @@ def forward(self, x: Union[Tensor, Proxy]) -> Union[Tensor, Proxy]:
...
+    def __init__(self,
+                 method: str = "max",
+                 kernel_size: int = 2,
+                 **kwargs):
+        super().__init__()
```
Comment (illian01) [inline src/netspresso_trainer/models/op/custom.py:180]: `MaxPool2d` and `AvgPool2d` have `kernel_size`, `stride`, `padding` parameters as common.
To clarity, I think `stride`, `padding` should also added as parameters.

- https://pytorch.org/docs/stable/generated/torch.nn.MaxPool2d.html
- https://pytorch.org/docs/stable/generated/torch.nn.AvgPool2d.html

**Comment by hglee98** [comment]
Thank you for the review! I've included your feedback from additional commits!🙇🏻‍♂️

**Review by illian01 — APPROVED** [review]
LGTM!
